docusaurus/docs/create-a-plugin/extend-a-plugin/instrument-a-backend-plugin.md

```diff
@@ -0,0 +1,421 @@
+
+### Implement metrics in your plugin
+
+[Grafana plugin SDK for Go](../../introduction/grafana-plugin-sdk-for-go.md) uses the [Prometheus instrumentation library for Go applications](https://github.com/prometheus/client_golang). Any custom metric registered with the [default registry](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#pkg-variables) will be picked up by the SDK and exposed through the [Collect metrics capability](../../introduction/backend.md#collect-metrics).
+
+For convenience, It's recommended to use the [promauto package](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promauto) when creating custom metrics since it will automatically register the metric in the [default registry](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#pkg-variables).
+
+**Example:**
+
+```go
+package plugin
+
+import (
+    "context"
+
+    "github.com/prometheus/client_golang/prometheus"
+    "github.com/prometheus/client_golang/prometheus/promauto"
+
+    "github.com/grafana/grafana-plugin-sdk-go/backend"
+)
+
+var queriesTotal = promauto.NewCounterVec(
+    prometheus.CounterOpts{
+        Namespace: "grafana_plugin",
+        Name:      "query",
+        Help:      "Total number of queries.",
+    },
+    []string{"query_type"},
+)
+
+func (ds *Datasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+    for _, q := range req.Queries {
+        queriesTotal.WithLabelValues(q.QueryType).Inc()
+    }
+}
+```
+
+### Best practices/recommendations
+
+- Consider using the namespace _grafana_plugin_ as that would prefix any defined metric names with plugin*. This will make it clear for operators that any metric named \*grafana_plugin*\* originates from a Grafana plugin.
+- Use snake case style when naming metrics, e.g. _http_request_duration_seconds_ instead of _httpRequestDurationSeconds_.
+- Use snake case style when naming metric labels, e.g. _status_code_ instead of _statusCode_.
+- If the metric type is a counter, name it with a _\_total suffix_, e.g. _http_requests_total_.
+- If the metric type is a histogram and you're measuring duration, name it with a _`_<unit>`_ suffix, e.g. _http_request_duration_seconds_.
+- If the metric type is a gauge, name it to denote it's a value that can increase and decrease , e.g. _http_request_in_flight_.
+
+#### Label values and high cardinality
+
+Be careful with what label values you add/accept. Using/allowing too many label values could result in high cardinality problems.
+
+If label values originate from user input they should be validated/cleaned. Use `metricutil.SanitizeLabelName("<label value>")` from pkg/infra/metrics/metricutil package to sanitize label names. Very important to only allow a predefined set of labels to minimize the risk of high cardinality problems.
```

Comment:
Room for improvement/possible to skip this. I think the SDK/Grafana should validate/sanitize to ease developer experience. Not sure it's done currently in https://github.com/prometheus/common/blob/6ff04000d9b52262fe36f82f51ab43caa45fb8e6/expfmt/text_create.go#L66 used by the SDK https://github.com/grafana/grafana-plugin-sdk-go/blob/24551aba3f95770be7b3c42f8403e92f449e48b5/backend/diagnostics_adapter.go#L34

Reply:
Avoid slashes except in code.

Reply:
Fixed